### PR TITLE
III-5220 - Add toggle to enable new entry form

### PR DESCRIPTION
**Pull request by simon-debruijn**
### Added

- Add toggle to enable/disable beta features

### Changed

- [Make base switch easier to style](https://github.com/cultuurnet/udb3-frontend/pull/615/commits/24385be67c5c7ef9c14f67441fee6d65f97024c2)
~~Refactor Button and Link components to not need `customChildren` and `shouldShowText` props~~

The refactoring of Button and Link broke the website's behaviour in several places, and the code changes for this PR were getting too big. We can do this refactor at another time in a separate PR.

---

Ticket: https://jira.uitdatabank.be/browse/III-5220

Screenshots:

**Desktop:**
Screenshot 2023-02-21 at 16 48 30

**Mobile:**
Screenshot 2023-02-21 at 16 48 59

### CODE REVIEW

File: src/hooks/useFeatureFlag.ts [hunk 1]
```diff
@@ -10,28 +18,56 @@ const createCookieName = (identifier: string) => `ff_${identifier}`;
...
+  return statePair;
 };
 
 const isFeatureFlagEnabledInCookies = (
   featureFlagName: FeatureFlagName,
   cookies: any,
 ) => {
   const cookieName = createCookieName(featureFlagName);
-  return cookies?.[cookieName] === 'true' ?? false;
+  return cookies?.[cookieName] === 'true' || false;
```
Comment (brampauwelyn) [inline src/hooks/useFeatureFlag.ts:70]: could be like this right? 
```suggestion
  return cookies?.[cookieName] === 'true'
```

**Review by brampauwelyn — COMMENTED** [review]
When the toggle has the focus state it seems to be cut of
Screenshot 2023-02-22 at 09 11 23

File: src/hooks/useFeatureFlag.ts (hunk 1 above)
Comment (simon-debruijn) [reply src/hooks/useFeatureFlag.ts:70]: Yes should be fine indeed 👍🏻

**Comment by simon-debruijn** [comment]
> @simon-debruijn Not sure if it's related but the online toggle now throws an error when you click on it.

I'm looking into the rendering errors. it has to do with the `customChildren` indeed. Something with `inner` in Button.

**Comment by simon-debruijn** [comment]
> The refactoring of Button and Link broke the website's behaviour in several places, and the code changes for this PR were getting too big. We can do this refactor at another time in a separate PR.

The refactoring of Button and Link broke the website's behaviour in several places, and the code changes for this PR were getting too big. We can do this refactor at another time in a separate PR.

**Comment by simon-debruijn** [comment]
> 

The refactoring of Button and Link broke the website's behaviour in several places, and the code changes for this PR were getting too big. We can do this refactor at another time in a separate PR.

**Comment by simon-debruijn** [comment]
> When the toggle has the focus state it seems to be cut of Screenshot 2023-02-22 at 09 11 23

This should be fixed now with the 2 latest commits regarding adding extra margin to account for box-shadow

**Pull request merged**